x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/geo/GeoProcessor.java

```diff
@@ -28,7 +28,10 @@ default R apply(Object o) {
     }
 
     public enum GeoOperation {
-        ASWKT(GeoShape::toString);
+        ASWKT(GeoShape::toString),
+        X(GeoShape::getX),
+        Y(GeoShape::getY),
+        GEOMETRY_TYPE(GeoShape::getGeometryType);
```

Comment:
Can you put this one second in line, please? (considering an alphabetic ordering)